### PR TITLE
#164536900 Login with the andela api

### PR DESCRIPTION
**Pull request by MuhanguziDavid**
**What does this PR do?**

- Ensures that the Andela API is used during login.
- Adds tests.

**Description of task to be completed?**
When a user clicks on the Login button, he is redirected to the Andela API which adds a token to the cookies. The token is used to determine user roles which are used to grant user's access. 

**How should this be manually tested?**
- Clone and set up the project using the instruction in the README.
- $ git checkout to `ft-use-andela-api-to-login-164536900` branch which contains the changes.
- Add the following variables to your frontend `.env` file: `REACT_APP_URL=http://localhost:3000/login` and `REACT_APP_ANDELA_AUTH_HOST=https://api-prod.andela.com`
- Use $ yarn start to start the application
- Start the frontend.
- Click on the login button.
You will get a `502 Bad Gateway` response.
- Open the app in an incognito window and login.
The going should be successful
- Inspect and check `cookies` under `Application`
There should be a cookie variable called `jwt-token`.
- Copy and paste the token in `https://jwt.io/` for it to be decoded.
The decoded token should include your user information which included `roles`.

**What are the relevant pivotal tracker stories?**
[#164536900](https://www.pivotaltracker.com/story/show/164536900)

### CODE REVIEW

File: src/components/Header/index.jsx [hunk 1]
```diff
@@ -67,7 +69,7 @@ class Header extends PureComponent {
           <div
             className="user-name"
             data-initials={initials}
-            data-name={name}
+            data-name={`${firstName}' '${lastName}`}
```
Comment (ssewilliam) [inline src/components/Header/index.jsx]: Hey @MuhanguziDavid, I do not think we need the single quotes here

**Review by ssewilliam — DISMISSED** [review]

**Comment by ssewilliam** [comment]
Please modify the *PR description* so that the reviewers can understand how to replicate the same setup for all the tests.

**Comment by MuhanguziDavid** [comment]
Thanks @ssewilliam, I have updated the PR description as requested.

File: src/components/Header/index.jsx (hunk 1 above)
Comment (MuhanguziDavid) [reply src/components/Header/index.jsx]: Thanks, @ssewilliam, I have removed the quotes.

**Comment by bozicschucky** [comment]
hey @MuhanguziDavid, fix the test coverage issue emerging from this PR.

**Comment by katunold** [comment]
Hey @MuhanguziDavid , I have tried to test manually using the steps you provided in the PR description but I didn't get the expected results. Could you kindly look into this.

> **Description of task to be completed?**
> When a user clicks on the Login button, he is redirected to the Andela API which adds a token to the cookies. The token is used to determine user roles which are used to grant user's access.
> 
> **How should this be manually tested?**
> 
> * Clone and set up the project using the instruction in the README.
> * $ git checkout to `ft-use-andela-api-to-login-164536900` branch which contains the changes.
> * Use $ yarn start to start the application
> * Start the frontend.
> * Click on the login button.
>   You will get a `502 Bad Gateway` response.
> * Open the app in an incognito window and login.
>   The going should be successful
> * Inspect and check `cookies` under `Application`
>   There should be a cookie variable called `jwt-token`.
> * Copy and paste the token in `https://jwt.io/` for it to be decoded.
>   The decoded token should include your user information which included `roles`.

**Comment by MuhanguziDavid** [comment]
> hey @MuhanguziDavid, fix the test coverage issue emerging from this PR.

I have worked on the test coverage @bozicschucky

**Comment by MuhanguziDavid** [comment]
@katunold, please elaborate on the results you got.

**Review by katunold — APPROVED** [review]
LGTM

**Comment by bozicschucky** [comment]
hey @MuhanguziDavid  the logout issue of this Pr. Thanks

**Comment by MuhanguziDavid** [comment]
Thank you, @bozicschucky, I have fixed the logout issue

**Review by bozicschucky — APPROVED** [review]
Tested it out and the functionality seems to be working as expected.

**Comment by MuhanguziDavid** [comment]
Conflicts have been resolved @ssewilliam

**Pull request merged**